src/printer.js

```diff
@@ -1455,15 +1479,17 @@ function genericPrintNoParens(path, options, print) {
       }
 
       return "";
+    case "TSTupleType":
     case "TupleTypeAnnotation":
+      let typesField = n.type === "TSTupleType" ? "elementTypes" : "types"
```

Comment:
I'm sorry, I don't understand. I don't mutate `n` here, at least not in a way I can see.
Could you elaborate?